csg.js

```diff
@@ -6481,6 +6531,11 @@ for solid CAD anyway.
                     }
                     thisside = sideTagToSideMap[nextsidetag];
                 } // inner loop
+                // due to the logic of CAG.fromPoints()
+                // move the first point to the last
+                if (connectedVertexPoints.length > 0) {
```

Comment:
Agreed. But let's save that for another discussion about what is a 100% proper CAG.